src/plugins/intel_gpu/src/plugin/transformations/fc_horizontal_fusion.cpp

```diff
@@ -300,6 +300,80 @@ FullyConnectedHorizontalFusion::FullyConnectedHorizontalFusion(bool fuse_mlp_swi
             }
             org_fc->clear_control_dependencies();
         }
+
+        // Merge scalar multiply layers into one when all scalar constants have the same value.
+        //
+        //          FusedFC                     FusedFC
+        //             |                           |
+        //       VariadicSplit      ==>         new_Mul  (to be fused with FusedFC)
+        //      /      |      \                    |
+        //    Mul     Mul     Mul            VariadicSplit
+        //     |       |       |             |     |     |
+        const auto is_scalar_const = [](const ov::Output<ov::Node>& output) -> bool {
+            if (!ov::is_type<ov::op::v0::Constant>(output.get_node()))
+                return false;
+            const auto shape = output.get_partial_shape();
+            if (shape.is_dynamic())
+                return false;
+            return ov::shape_size(shape.to_shape()) == 1;
+        };
+
+        auto get_const_value = [](const std::shared_ptr<ov::op::v0::Constant>& const_layer) -> float {
+            float const_value = -1.f;
+            if (const_layer->get_element_type() == ov::element::f16) {
+                const_value = std::stof(const_layer->get_data_ptr<ov::float16>()->to_string());
+            } else if (const_layer->get_element_type() == ov::element::f32) {
+                const_value = *const_layer->get_data_ptr<float>();
+            }
+            return const_value;
+        };
+
+        float const_value = -1.f;
```

Comment:
Probably it makes sense to use some other default value? Something like `std::numeric_limits<float>::max()` or independent flag instead of `const_value < 0.f` checks, just to support a wider range of scalars?

Reply:
I'd say that it's better to use separate flag that indicates if transform shall be done